### PR TITLE
Add crowdfund app create endpoint

### PR DESCRIPTION
**Pull request by bolatovumar**
Added Greenfield API `/api/v1/stores/{storeId}/apps/crowdfund` endpoint for creating crowdfund apps.

Address #3458

### CODE REVIEW

File: BTCPayServer.Client/Models/CreateAppRequest.cs [hunk 1]
```diff
@@ -1,8 +1,66 @@
+using System;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
 
 namespace BTCPayServer.Client.Models
 {
+    class DateTimeJsonConverter : JsonConverter
```
Comment (Kukks) [inline BTCPayServer.Client/Models/CreateAppRequest.cs:7]: use `NBitcoin.JsonConverters.DateTimeToUnixTimeConverter` instead

File: BTCPayServer.Client/Models/CreateAppRequest.cs [hunk 2]
```diff
@@ -38,4 +96,44 @@ public class CreatePointOfSaleAppRequest : CreateAppRequest
...
+        Year
+    }
+
+    public class CreateCrowdfundAppRequest : CreateAppRequest
+    {
+        public string Title { get; set; } = null;
+        public bool? Enabled { get; set; } = null;
+        public bool? EnforceTargetAmount { get; set; } = null;
+        [JsonConverter(typeof(DateTimeJsonConverter))]
+        public DateTime? StartDate { get; set; } = null;
```
Comment (Kukks) [inline BTCPayServer.Client/Models/CreateAppRequest.cs:115]: Ideally, we stick to `DateTimeOffset` in Greenfield to stay consistent.

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer.Client/Models/CreateAppRequest.cs (hunk 1 above)
Comment (Tastry) [reply BTCPayServer.Client/Models/CreateAppRequest.cs:7]: ?

Comment (bolatovumar) [reply BTCPayServer.Client/Models/CreateAppRequest.cs]: Done here: https://github.com/btcpayserver/btcpayserver/pull/4068/commits/affa2eaf8dfe8fef48f2e6d7c69076e355d838ca

File: BTCPayServer.Client/Models/CreateAppRequest.cs (hunk 2 above)
Comment (bolatovumar) [reply BTCPayServer.Client/Models/CreateAppRequest.cs]: Done here: https://github.com/btcpayserver/btcpayserver/pull/4068/commits/fd4f64605ec148e7cf6f400c38394cb471f3f973

File: BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs [hunk 3]
```diff
@@ -143,6 +175,43 @@ private IActionResult AppNotFound()
             return this.CreateAPIError(404, "app-not-found", "The app with specified ID was not found");
         }
 
+        private CrowdfundSettings ToCrowdfundSettings(CreateCrowdfundAppRequest request)
+        {
+            var parsedSounds = ParseBySplitByComma(request.Sounds);
+            var parsedColors = ParseBySplitByComma(request.AnimationColors);
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs:181]: why using csv rather than just an array?

Comment (bolatovumar) [reply BTCPayServer/Controllers/GreenField/GreenfieldAppsController.cs]: @NicolasDorier updated here: https://github.com/btcpayserver/btcpayserver/pull/4068/commits/5d328ab392c2651060305c6a15d06e569bf3d84d

**Comment by bolatovumar** [comment]
Failing test unrelated

Capture

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json [hunk 4]
```diff
@@ -360,6 +432,163 @@
...
+                        "default": false,
+                        "nullable": true
+                    },
+                    "startDate": {
+                        "type": "number",
+                        "description": "UNIX timestamp for crowdfund start time (https://www.unixtimestamp.com/)",
+                        "example": 768658369,
+                        "nullable": true
+                    },
+                    "endDate": {
```
Comment (Kukks) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:475]: follow this example:
```
 "createdAt": {
                        "description": "The unix timestamp when the payment got created",
                        "nullable": true,
                        "allOf": [ {"$ref": "#/components/schemas/UnixTimestamp"}]
                    },
```

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json (hunk 4 above)
Comment (bolatovumar) [reply BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:476]: @Kukks updated in https://github.com/btcpayserver/btcpayserver/pull/4068/commits/b8fa4cfd5905ac75b1e063eb21521a43314287e0

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json [hunk 5]
```diff
@@ -360,6 +432,165 @@
...
+                        "default": false,
+                        "nullable": true
+                    },
+                    "sortPerksByPopularity": {
+                        "type": "boolean",
+                        "description": "Sorts perks by popularity if set to true",
+                        "default": false,
+                        "nullable": true
+                    },
+                    "sounds": {
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:582]: it's not comma separated, this is an array

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json [hunk 6]
```diff
@@ -360,6 +432,165 @@
...
+                        "type": "string",
+                        "description": "Comma-separated list of custom sounds to use on new contributions",
+                        "nullable": true
+                    },
+                    "animationColors": {
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:587]: it's not comma separated, this is an array

Comment (bolatovumar) [reply BTCPayServer/wwwroot/swagger/v1/swagger.template.apps.json:598]: Updated in https://github.com/btcpayserver/btcpayserver/pull/4068/commits/26a1bfb6172bf1a77228a8677fd4e277ee322284

**Pull request merged**